### ISSUE TITLE
[Security] leak of database password forced

### ISSUE THREAD

**Issue by genofire** [documize/community#243]
It is a bad idea to have the only way of configuration the database the by progress argument.
On this way every user on the same host could explorer the username and password by just the command:
`ps -aux`

**Comment by HarveyKandola** [comment]
True. You can also use environment variables but that just moves the problem and still fails to ```ps -aux```.

We've yet to select a good cross-platform (macOS, Linux, Windows) and cross-database solution (SQL Server, MySQL, PostgreSQL...).

**Comment by genofire** [comment]
^^ A small config in toml/yaml format ??

**Comment by HarveyKandola** [comment]
You would probably still want to encrypt the YAML file and then handle decryption during Documize start-up...

**Comment by genofire** [comment]
No, just read it from the configfile, just like every else program.

You let manage the permission of reading the password by the operation system, which has a managment of permission to get access for files.

In the end you make:
`chown documize:documize /etc/documize.conf; chmod 0400 /etc/documize.conf`

**Comment by HarveyKandola** [comment]
All sounds good from Linux/Darwin perspective, we just need to consider cross-platform support, specifically Windows. 

Things like service/daemon accounts usually cause people issues. Clear instructions required!

The ```documize.conf``` location needs to be cross-platform compliant. Support burden increases when configs get picked up without the user knowing they even existed.

FYI, we also did consider https://www.vaultproject.io -- might clash with existing secrets management tooling that people already use internally.

**Comment by genofire** [comment]
Okay, would a clear instruction like this help?

**Step 1:**
Create a config file:
```toml
salt = "1234567890acbdef"
location = "selfhost"
site_mode = "1"

[http]
ssl_cert = ""
ssl_key = ""
port = "5001"
port_forcessl = true

[database]
type = "mysql"
connection = "documize:password@tcp(localhost:3306)/documize"
```
At a place, which you like to store it.
On windows maybe unter: `%ProgramData%/documize.conf`
On unixiod maybe under: `/etc/documize.conf`
Make sure this config file is only readable from necessary users.
(on unixiod fast with: `chown documize:documize /etc/documize.conf; chmod 0400 /etc/documize.conf`)

**Step 2:**
start `documize -configfile=/etc/documize .conf` as deamon
or create a reboot save service for it [windows](https://docs.gitea.io/en-us/windows-service/) or [unixiod](https://docs.gitea.io/en-us/linux-service/)

**Alternative**
Just for testing there is possible to call documize with arguments  instatt of configfile:
`documize  -port=5001 -db='documize:password@tcp(localhost:3306)/documize' -dbtype=mysql -salt=1234567890acbdef`

---

Nice that you manage your secrets safe and that you have only running this service at one "host" (that feels me really save). But for some smaller hoster which would not like to waste ressourcen by containers or virtuelle maschines have to manage multiple services on one maschine ...

**Comment by HarveyKandola** [comment]
Makes sense -- we're looking into this right now...

**Comment by HarveyKandola** [comment]
This has been done for next release.

```
.\documize runtime.conf
```

**Comment by sauls8t** [comment]
Released in v2.4.2:

In addition to specifying parameters in the command line and environment variables, you can now omit all parameters and provide a config file. Example:

```
./documize myconfig.conf
```

Documize will look for a documize.conf config file if no other arguments are provided. Example:

```
./documize
```
